examples/csv-io/csv_io.bal

```diff
@@ -46,31 +46,31 @@ public function main() {
     // ./files/sample.csv for later use.
     // The record separator of the `.CSV` file is a
     // new line, and the field separator is a comma (,).
-    io:CSVChannel csvChannel = io:openCsvFile(srcFileName, mode = "w");
+    io:WritableCSVChannel wCsvChannel = io:openWritableCsvFile(srcFileName);
     string[][] data = [["1", "James", "10000"], ["2", "Nathan", "150000"],
     ["3", "Ronald", "120000"], ["4", "Roy", "6000"], ["5", "Oliver", "1100000"]];
-    writeDataToCSVChannel(csvChannel, ...data);
-    closeCSVChannel(csvChannel);
+    writeDataToCSVChannel(wCsvChannel, ...data);
+    closeWritableCSVChannel(wCsvChannel);
```

Comment:
Ideally, we should have a single close function to handle both readable and writable channels. WDYT?

Reply:
Yes we considered that and your argument is valid. For this to be generic we need to have an expression as the following
```
    match csvchannel{
        object{public function close() returns error|();} ch =>{}
        error channelCloseError => {
            log:printError("Error occured while closing the channel: ",
                err = channelCloseError);
        }
        () => io:println("CSV channel closed successfully.");
    }
```
There's a limitation is the match command which does not match against objects which has private fields, which is a bug. based on the discussion we had with @SupunS. Hence we decided to introduce multiple functions instead.